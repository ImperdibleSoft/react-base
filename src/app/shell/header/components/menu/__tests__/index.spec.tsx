import * as React from 'react';
import { shallow } from 'enzyme';

import Menu from '..';

describe('Menu', () => {
  it('should render a valid closed menu', () => {
    const component = shallow(
      <Menu handleClick={() => {}} isOpen={false} toggler={{ label: 'test' }}>
        <ul>
          <li>Test</li>
        </ul>
      </Menu>
    );

    expect(component).toMatchSnapshot();
  });

  it('should render a valid open menu', () => {
    const component = shallow(
      <Menu handleClick={() => {}} isOpen={true} toggler={{ label: 'test' }}>
        <ul>
          <li>Test</li>
        </ul>
      </Menu>
    );

    expect(component).toMatchSnapshot();
  });

  it('should render a valid open menu with icon', () => {
    const component = shallow(
      <Menu handleClick={() => {}} isOpen={true} toggler={{ icon: 'test', label: 'test' }}>
        <ul>
          <li>Test</li>
        </ul>
      </Menu>
    );

    expect(component).toMatchSnapshot();
  });
});
