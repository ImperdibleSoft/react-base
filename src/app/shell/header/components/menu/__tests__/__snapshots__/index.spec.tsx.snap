// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu should render a valid closed menu 1`] = `
<Wrapper
  isRight={true}
  onClick={[Function]}
>
  <Toggler>
    <Content
      options={
        Object {
          "iconLast": true,
          "label": "test",
        }
      }
    />
  </Toggler>
</Wrapper>
`;

exports[`Menu should render a valid open menu 1`] = `
<Wrapper
  isRight={true}
  onClick={[Function]}
>
  <Toggler>
    <Content
      options={
        Object {
          "iconLast": true,
          "label": "test",
        }
      }
    />
  </Toggler>
</Wrapper>
`;

exports[`Menu should render a valid open menu with icon 1`] = `
<Wrapper
  isRight={true}
  onClick={[Function]}
>
  <Toggler>
    <Content
      options={
        Object {
          "icon": "test",
          "iconLast": true,
          "label": "test",
        }
      }
    />
  </Toggler>
</Wrapper>
`;
